spec.md

```diff
@@ -406,18 +404,28 @@ See
 [CloudEvent Attributes Extensions](primer.md#cloudevent-attribute-extensions)
 for additional information concerning the use and definition of extensions.
 
-This specification places no restriction on the type or semantics of the
-extension attributes. Each definition of an extensions SHOULD fully define all
-aspects of the attribute - e.g. its name, semantic meaning and possible values
-or even to indicate that it places no restrictions on its values. New extension
-definitions SHOULD use a name that is descriptive enough to reduce the chances
-of name collisions with other extensions. In particular, extension authors
-SHOULD check the [documented extensions](documented-extensions.md) document for
-the set of known extensions - not just for possible name conflicts but for
-extensions that might be of interest.
-
-Each specification that defines how to serialize a CloudEvent will define how
-extension attributes will appear.
+Extension attributes MUST conform to one of the types in the [Type
+System](#type-system).  This specification does not otherwise constrain the
+content or semantics of extension attributes.
+
+The definition of an extension attribute SHOULD define its [type](#type-system),
+name, semantic meaning and constraints on legal values.
+
+New extension definitions SHOULD use a name that is descriptive enough to reduce
+the chances of name collisions with other extensions. In particular, extension
+authors SHOULD check the [documented extensions](documented-extensions.md)
+document for the set of known extensions - not just for possible name conflicts
+but for extensions that might be of interest.
+
+The consumer of an extension attribute MUST know the extension definition to use
```

Comment:
I don't think we can make this a MUST since 1) we can't enforce or check it, and 2) a "consumer" doesn't actually need to know about it if all it's doing is passing it along. I tried to reword it to keep the gist of what you're trying to say, but I can't seem to find a wording that doesn't seem obvious.....e.g. "if a consumer wants to successfully process an extension then it needs to know about it" feels silly  :-)

What is the requirement you're trying to make/enforce here?

Reply:
Agreed the MUST is wrong. I'm trying to say - extension attributes can't be relied on to carry type information so the consumer has to Just Know from somewhere else. They can't write code like `if attr[foo].isa timestamp...` they have to know to say `ts := AsTimestamp(attr[foo])`

Reply:
Oops, already fixed.